src/features/Admin/Ads/AddNewAd/AddNewAd.tsx

```diff
@@ -0,0 +1,186 @@
+import React, { useContext, useEffect, useState } from 'react';
+import { AuthContext } from '../../../../context/AuthContext';
+import { useNavigate } from 'react-router-dom';
+import { IAds } from '../../../../interface/AdsInterface';
+import axios from 'axios';
+import { addAdsUrl, adsUrl, roomsUrl } from '../../../../services/api';
+import { SubmitHandler, useForm } from 'react-hook-form';
+import { Button, FormControl, Grid, InputLabel, MenuItem, Paper, TextField } from '@mui/material';
+import { Box, Container } from '@mui/system';
+import style from './AddNewAd.module.scss';
+import { Select } from '@mui/material';
+import { toast } from 'react-toastify';
+const AddNewAd: React.FC = () => {
+    const { requestHeaders } = useContext(AuthContext);
+    const navigate = useNavigate();
+
+    const [roomsList, setRoomsList] = useState([]);
+    const [isLoading, setIsLoading] = useState(false);
+    const [active, setActive] = useState('');
+
+    const handleChange = (event: React.ChangeEvent<{ value: unknown }>) => {
+        // setActive(event.target.value as string);
+        setActive(event.target.value);
+    };
+    const {
+        register,
+        handleSubmit,
+        setValue,
+        getValues,
+        formState: { errors },
+    } = useForm<IAds>();
+
+    // Format Data
+    // const appendToFormData = (data: IAds) => {
+    //     const formData = new FormData();
+    //     formData.append('room', data.room || ''); // Ensure data.room is not undefined
+    //     formData.append('discount',data.discount || 0); // Convert discount to string
+    //     formData.append('isActive', data.isActive); // Convert isActive to string
+    //     return formData;
+    // };
+
+    // Cancel Button
+    const goBack = () => {
+        navigate(-1);
+    };
+    const getAllAds = () => {
+        axios.get(`${adsUrl}`, {
+            headers: requestHeaders
+        })
+            .then((response) => {
+                // setAdsList(response.data.data.ads)
+            })
+            .catch((error) => {
+              
+            })
+    }
+
+
+    // Create New Room
+    const onSubmit: SubmitHandler<IAds> = async (data: IAds) => {
+        // const isActiveBoolean = active === 'yes';
+        // data.isActive = isActiveBoolean;
+        data.isActive = active === 'yes';
+        axios
+            .post(`${addAdsUrl}`, data, { headers: requestHeaders })
+            .then((response) => {
+                getAllAds();
```

Comment:
We don't need to call `getAllAds` here, as just navigate to `home/ads` will mount the ads component, and in the `useEffect`, we get  all ads when the component first mounts.
So you can remove all logic related to `getAllAds` from this component.